src/mds/FSMap.cc

```diff
@@ -491,21 +491,21 @@ void Filesystem::decode(bufferlist::iterator& p)
 
 int FSMap::parse_filesystem(
       std::string const &ns_str,
-      std::shared_ptr<const Filesystem> *result
+      std::shared_ptr<const Filesystem> &result
```

Comment:
Using pointers for output arguments is a style guide thing (https://github.com/ceph/ceph/blob/master/CodingStyle#L127)